src/nbl/video/IGPUDescriptorSet.cpp

```diff
@@ -124,6 +124,33 @@ void IGPUDescriptorSet::processWrite(const IGPUDescriptorSet::SWriteDescriptorSe
     incrementVersion();
 }
 
+void IGPUDescriptorSet::dropDescriptors(const IGPUDescriptorSet::SDropDescriptorSet& drop, const asset::IDescriptor::E_TYPE type)
+{
+    assert(drop.dstSet == this);
+
+	auto* dstDescriptors = drop.dstSet->getDescriptors(type, drop.binding);
+	auto* dstSamplers = drop.dstSet->getMutableSamplers(drop.binding);
+	assert(dstDescriptors);
+    // Samplers are only used in the combined image sampler descriptors
+    if (type == asset::IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER)
+	    assert(dstSamplers); 
+
+	if (dstDescriptors)
```

Comment:
can;t have it both ways, either assert and don't if `dstDescriptors`, or if and don't assert